### PR TITLE
SONAR-6385 add WS to list SonarQube upgrades

### PR DESCRIPTION
**Pull request by sns-seb**
I'll investigate why I have imports reodered on my laptop before merging to master

### CODE REVIEW

**Comment by simonbrandhof** [comment]
Terminology feedback - api/plugins/updates is the similar WS for plugins. For consistency don't you think that  the same action should be used: "updates" or "upgrades" ?

File: server/sonar-server/src/main/java/org/sonar/server/platform/ws/UpgradesSystemWsAction.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      .setResponseExample(Resources.getResource(this.getClass(), "example-updates_plugins.json"));
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    JsonWriter jsonWriter = response.newJsonWriter().setSerializeEmptys(false);
+
+    jsonWriter.beginObject();
+    UpdateCenter updateCenter = updateCenterFactory.getUpdateCenter(DO_NOT_FORCE_REFRESH);
+    writeSonarUpgrades(jsonWriter, updateCenter);
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ws/UpgradesSystemWsAction.java]: sonar is now sonarqube :-) Method could be renamed "writeSystem*"

**Comment by simonbrandhof** [comment]
else LGTM

**Comment by sns-seb** [comment]
> Terminology feedback - api/plugins/updates is the similar WS for plugins. For consistency don't you think that the same action should be used: "updates" or "upgrades" ?

In plugin WebService, when talking about the plugin compatibility/update status, we distinguish  "plugin updates" from "SQ updates" by calling the later "upgrades". 

In addition, I like the term "upgrade" for the system, since imo the noun implies a bigger modification than a "simple" update. Upgrading SQ may involve updating plugins, it makes sense. The Oxford definition of the nouns kind of goes in my direction (but is subject to interpretation) http://www.oxforddictionaries.com/definition/english/upgrade http://www.oxforddictionaries.com/definition/english/update.

**Comment by simonbrandhof** [comment]
Good, you have a strong opinion. So ok !

**Pull request merged**